src/alitra/frame_dataclasses.py

```diff
@@ -104,6 +109,33 @@ def from_array(
         )
 
 
+@dataclass
+class Quaternion:
+    frame: Literal["robot", "asset"]
+    x: float = 0
+    y: float = 0
+    z: float = 0
+    w: float = 1
```

Comment:
Selected x, y, z, w as the representation as these were used in Echo. Open to opinions though as it is not the one I would have selected by default.

w is the scalar.